### PR TITLE
feat: survey error modal/page

### LINKED ISSUE #1082: [FEATURE][FormTribe 🔥][500 Points] Add Error handling modal for Link & in-App Surveys
### Welcome to the FormTribe

This issue will get you the points mentioned in issue title for the [FormTribe Hackathon](https://formtribe.com)

**Preparation:**
If this is your first time contributing, please read:

- Our [introductory blog post](https://formbricks.com/blog/join-the-formtribe)
- Overview of [Code Guidelines at Formbricks.](https://formbricks.com/docs/contributing/how-we-code)

**Only when your PR follows _all_ of the best practices it will get merged.**

Want more points quick? Complete [Side Quests 🏰 ](https://formbricks.com/formtribe#side-quests)

--------

### Is your feature request related to a problem? Please describe.

We introduced optimistic response updates where we queue responses and try to send them in the background without waiting with every question submit.

We retry to resend all response updates up to three times when an error occurs. But when a user finishes the survey and we are still unable to send the responses, the user sees an error modal.

### Describe the solution you'd like

Solution: Please build an error screen that shows all the responses of the user together with the questions plus an option to contact the survey owner.

Tasks

1. In the Team Settings, add a new card above Delete Team:

image
image

2. We'll build an error component which is displayed, when the submission failed. To do so, add a new element to the survey package at packages/surveys/components. The component looks as follows:

image

- The UI should be self-explanatory :) The Copy your feedback button copies the feedback in a format which makes sense and the Send via Email button opens a mailto: link with the public support email as the recipient
- If the public support email is not set, the "&" and the "Send via email" button isn't displayed.

### UX Notes

- For in-app surveys, we want the modal to close while the data is attempted to submit. Only after it failed a third time, we reopen the modal and display the error component.
- For link surveys, we'll show the Thank You card and replace it with the error component, once the error is thrown.

### PR DESCRIPTION
**Pull request by ty-kerr**
## What does this PR do?

Adds an error page and error modal for link surveys and in-app surveys respectively for when submission of responses have failed more than the number of retries alloted. The error page/modal shows all the respones from the survey and has two button, Copy all response as json, and send to email. 

Fixes #1082

https://www.loom.com/share/a04569b18f8e4bf8bcf402f9bfb2611e

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?

Create a link survey or in-app survey, Mock response submission failure in responseQueue.ts, by fixing success to false on line 46. Complete the survey and wait 3 seconds after the thank you page is shown. The error page should appear

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [x] If a UI change was made: Added a screen recording or screenshots to this PR
- [] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: packages/surveys/src/components/OpenTextQuestion.tsx [hunk 1]
```diff
@@ -45,7 +45,7 @@ export default function OpenTextQuestion({
       onSubmit={(e) => {
         e.preventDefault();
         //  if ( validateInput(value as string, question.inputType, question.required)) {
-        onSubmit({ [question.id]: value, inputType: question.inputType });
```
Comment (ty-kerr) [inline packages/surveys/src/components/OpenTextQuestion.tsx:48]: remove this as it was added unused fields to the response accumulator

**Comment by jobenjada** [comment]
Works well :) 

@pandeymangg please review

File: packages/types/v1/teams.ts [hunk 2]
```diff
@@ -7,12 +7,14 @@ export const ZTeam = z.object({
   name: z.string(),
   plan: z.enum(["free", "pro"]),
   stripeCustomerId: z.string().nullable(),
+  supportEmail: z.string().nullable(),
```
Comment (pandeymangg) [inline packages/types/v1/teams.ts:10]: please use `z.string().email().nullable()` 🙏

File: packages/types/v1/teams.ts [hunk 3]
```diff
@@ -7,12 +7,14 @@ export const ZTeam = z.object({
...
 });
 
 export const ZTeamUpdateInput = z.object({
   name: z.string(),
   plan: z.enum(["free", "pro"]).optional(),
   stripeCustomerId: z.string().nullish(),
+  supportEmail: z.string().nullish(),
```
Comment (pandeymangg) [inline packages/types/v1/teams.ts:17]: please use `z.string().email().nullish()` 🙏

**Review by pandeymangg — CHANGES_REQUESTED** [review]
Thanks a lot for the PR @ty-kerr. Looks and works great! 🙌  The code also looks great!
Please also add the loading state for the `EditTeamSupportEmail` component in `apps/web/app/(app)/environments/[environmentId]/settings/members/loading.tsx` 😊 and use the `email` type for supportEmail as I've mentioned below.
Thanks a lot!

**Comment by ty-kerr** [comment]
@pandeymangg 
Done with the changes from review and merged with main and resolved all conflicts. XD

**Review by mattinannt — APPROVED** [review]
@ty-kerr Thanks again for the PR and your great work on this feature 💪🔥🚀

As you may have noticed, we've had some internal discussions about the feature, it's scope and how it fits into our current Formbricks strategy.

We've decided to redesign the feature a bit; remove the support email address to have all feedback sent to the Formbricks platform and use the error modal to signal the user that their feedback can't be sent (probably because their internet connection isn't working).

(We are working on another feature to increase the availability of Formbricks Cloud, which makes the failure to send from our side very unrealistic.)

I have now made all the changes for this and we are ready to merge 💪 Sorry for the big delay 🙈

Screenshot 2024-01-22 at 16 36 15

**Pull request merged**